service/bfg/bfg.go

```diff
@@ -317,6 +317,92 @@ func (s *Server) handleBitcoinBalance(ctx context.Context, bbr *bfgapi.BitcoinBa
 	}, nil
 }
 
+func (s *Server) bitcoinBroadcastWorker(ctx context.Context) {
+	log.Tracef("bitcoinBroadcastWorker")
+	defer log.Tracef("bitcoinBroadcastWorker exit")
+
+	defer s.wg.Done()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+
+		serializedTx, err := s.db.BtcTransactionBroadcastRequestGetNext(ctx)
+		if err != nil {
+			log.Errorf("error getting next broadcast request: %v", err)
+			continue
+		}
+
+		// if we found NO txs to broadcast, wait a bit to check for more
+		if serializedTx == nil {
+			select {
+			case <-ctx.Done():
+				return
+			case <-time.After(5 * time.Second):
+				continue
+			}
+		}
+
+		rr := bytes.NewReader(serializedTx)
+		mb := wire.MsgTx{}
+		if err := mb.Deserialize(rr); err != nil {
+			log.Errorf("failed to deserialize tx: %v", err)
+			continue
+		}
+
+		var tl2 *pop.TransactionL2
+		for _, v := range mb.TxOut {
+			tl2, err = pop.ParseTransactionL2FromOpReturn(v.PkScript)
+			if err == nil {
+				break // Found the pop transaction.
+			}
+		}
+
+		if tl2 == nil {
+			log.Errorf("could not find pop tx")
+			continue
+		}
+
+		publicKeyUncompressed, err := pop.ParsePublicKeyFromSignatureScript(mb.TxIn[0].SignatureScript)
+		if err != nil {
+			log.Errorf("could not parse public key from signature script: %v", err)
+		}
+
+		hash := mb.TxHash()
+
+		if err := s.db.PopBasisInsertPopMFields(ctx, &bfgd.PopBasis{
+			BtcTxId:             hash[:],
+			BtcRawTx:            database.ByteArray(serializedTx),
+			PopMinerPublicKey:   publicKeyUncompressed,
+			L2KeystoneAbrevHash: tl2.L2Keystone.Hash(),
+		}); err != nil {
+			log.Errorf("error occurred inserting pop basis: %s", err)
+			continue
+		}
+
+		_, err = s.btcClient.Broadcast(ctx, serializedTx)
+		if err != nil {
+			log.Errorf("broadcast tx: %s", err)
```

Comment:
if we fail here, let's set the database record for an "immediate retry".  based on the rules this would take N minutes to retry.